src/main/java/com/purbon/kafka/topology/actions/topics/builders/TopicConfigUpdatePlanBuilder.java

```diff
@@ -35,30 +35,33 @@ public TopicConfigUpdatePlan createTopicConfigUpdatePlan(Topic topic, String ful
         .forEach(
             (configKey, configValue) -> {
               ConfigEntry currentConfigEntry = currentConfigs.get(configKey);
-              //TODO: Must analyze further if isDynamicTopicConfig is the correct thing here
-              if (!isDynamicTopicConfig(currentConfigEntry) && !currentConfigEntry.value().equals(configValue)) {
+              // TODO: Must analyze further if isDynamicTopicConfig is the correct thing here
+              // From browsing the code, it appears that all topic configs that are not default will
+              // be dynamic.
+              if (!isDynamicTopicConfig(currentConfigEntry)
+                  && !currentConfigEntry.value().equals(configValue)) {
                 topicConfigUpdatePlan.addNewConfig(configKey, configValue);
               } else if (!currentConfigEntry.value().equals(configValue)) {
-                  topicConfigUpdatePlan.addConfigToUpdate(configKey, configValue);
+                topicConfigUpdatePlan.addConfigToUpdate(configKey, configValue);
               }
-
```

Comment:
This block would not delete anything when inside the loop for existing configs. Moved it outside.

Reply:
Ja her har det gått litt for kjapt!